### PR TITLE
Added option for custom environment variables for PostgreSQL.

### PR DESCRIPTION
**Pull request by bokysan**
This commit enables the user to specify additional environment
variables in the stable/postgres chart.

**What this PR does / why we need it**:
This is especially important for custom Postgres images which might have different environment variables for different options or for enabling additional features.

**Special notes for your reviewer**:
Nothing special here, really, the same pattern for adding variables is also used in other charts.

### CODE REVIEW

**Comment by cpanato** [comment]
@bokysan please bump the chart version

**Comment by bokysan** [comment]
Version bumped.

File: stable/postgresql/values.yaml
Comment (cpanato) [inline stable/postgresql/values.yaml]: `extraEnv ` should be used and also please add this new variable in the readme.md

File: stable/postgresql/Chart.yaml
Comment (cpanato) [inline stable/postgresql/Chart.yaml]: would say to bump to `1.0.0` since this is a chart in stable folder and we had a call in the charts meeting this week to move all stable charts that are using version 0.x.x to be at least 1.0.0 for more info you can check here: https://docs.google.com/document/d/1h6UTTuNRbFI81higrN3JUV2XxyzqqVjZET4Xz4WTR-8/edit#heading=h.57pbxthvt6k3

**Review by cpanato — CHANGES_REQUESTED** [review]

File: stable/postgresql/values.yaml
Comment (bokysan) [reply stable/postgresql/values.yaml]: > extraEnv should be used and also please add this new variable in the readme.md

Sure, we could do that, but seems non-consistent. Other charts in this same repository use the `env` syntax, e.g.
https://github.com/helm/charts/blob/4dd8cdc6bde7701c96ceb171f3ac3e2c9f521efd/stable/sentry/templates/web-deployment.yaml#L99

Is there an official policy on this?

Comment (cpanato) [reply stable/postgresql/values.yaml]: can use env or extraEnv, but since this is extra variables I would prefer to use that. 
I was searching the docs and did not find, but one of the maintainers of the charts reviewed my PR some days ago and asked to use extraEnv as well: https://github.com/helm/charts/pull/7126#discussion_r210263817

Comment (bokysan) [reply stable/postgresql/values.yaml]: Sure, let's go with `extraEnv`then. I'd suggest reviewing other charts and setting a policy then to have everything in official repo as consistent as possible.

File: stable/postgresql/Chart.yaml
Comment (bokysan) [reply stable/postgresql/Chart.yaml]: I've updated chart version to `1.0.0`. While on the subject, I've also updated PostgreSQL version to `10.5`.

Comment (cpanato) [reply stable/postgresql/Chart.yaml]: please update the README.md as well

File: stable/postgresql/values.yaml
Comment (cpanato) [reply stable/postgresql/values.yaml]: yes agree. cc @davidkarlsen @mattfarina @unguiculus

File: stable/postgresql/values.yaml
Comment (desaintmartin) [inline stable/postgresql/values.yaml]: You are changing the postgres version here, which seems out of the scope of this PR.

Comment (bokysan) [reply stable/postgresql/values.yaml]: I agree. But then again so should changing the verison of the helm chart.

Happy to revert, but frankly:
- I'm not sure anymore what should be the scope and what not
- I have tested the chart with this new version and does work OK.

I'd kindly ask that this PR finally gets merged, since it works and it's an improvement given current situation either way.

Comment (desaintmartin) [reply stable/postgresql/values.yaml]: It would be healthier to separate concerns and do two differents PRs. A lot of people are cherry-picking commits and may see their postgres version change without realizing (no hint from commit message).

Although I am happy with both changes. :)

Comment (bokysan) [reply stable/postgresql/values.yaml]: What if we just change the commit message then?

**Comment by bokysan** [comment]
Hello gentlemen,

There were some comments from @cpanato, which were resolved. We had a discussion about raising a postgres version with @desaintmartin, which I would leave as is (and maybe fix the commit message).

What do we do about this pull request now? Any possibility of merging it?

File: stable/postgresql/values.yaml
Comment (desaintmartin) [reply stable/postgresql/values.yaml]: I'm personally against this, but let's see other people's point of view. It is easy to do two different PRs for different concerns.

**Comment by cpanato** [comment]
@bokysan talking with @desaintmartin the bump the Postgres version will be good in another PR. it is just my opinion as re-thinking this bump. Since this is a bump to a major version of Postgres maybe will be good to have in another pr

**Comment by bokysan** [comment]
I have rolled back the PostgreSQL version upgrade change.

I'll open a separate PR once this one gets merged.

**Comment by cpanato** [comment]
@bokysan rebase and lets try to get this merged!

**Comment by bokysan** [comment]
> @bokysan rebase and lets try to get this merged!

I'm sorry, I don't follow -- this has been rebased into one commit from the start and with every change we did. Everything is in commit 19b6e55751a2c12fce08f9a3be9820211ab0f19d. Do you need me to do anything else?

**Comment by cpanato** [comment]
@bokysan you need to rebase and then bump the chart version

**Comment by bokysan** [comment]
> @bokysan you need to rebase 
I've rebased everything ontho charts:master.

> and then bump the chart version
Version has already been bumped to 1.0.0 since Sep 6, 2018 at your request on that day.

**Comment by bokysan** [comment]
Hi,

@desaintmartin, @cpanato. I've reverted the rebase onto master. As we apparently don't understand each other, could you kindly spell it out for me (or direct me to a web page) what exactly you need me to do to get this merged?

Thank you,
B

**Comment by desaintmartin** [comment]
You should do a "rebase", i.e
`git fetch upstream && git rebase upstream/master && git push -f origin master`, `upstream` being this repository and `origin` your fork.
https://git-scm.com/book/en/v2/Git-Branching-Rebasing
The goal is for your branch to be up-to-date AND to have only your changes in this PR, nothing else.

**Comment by cpanato** [comment]
yeah, you should do a rebase, usually I add the `-i` to make interactive `git rebase -i upstream/master`

**Comment by bokysan** [comment]
Ok, done. I don't know how I was able to pull all the changes in the last time.

**Pull request merged**
